payment/src/main/java/com/yas/payment/controller/PaymentProviderController.java

```diff
@@ -1,22 +1,48 @@
 package com.yas.payment.controller;
 
 import com.yas.payment.service.PaymentProviderService;
-import com.yas.payment.viewmodel.PaymentProviderVm;
+import com.yas.payment.viewmodel.paymentprovider.CreatePaymentVm;
+import com.yas.payment.viewmodel.paymentprovider.PaymentProviderVm;
+import com.yas.payment.viewmodel.paymentprovider.UpdatePaymentVm;
+import jakarta.validation.Valid;
 import java.util.List;
-import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
-@RequiredArgsConstructor
 public class PaymentProviderController {
+
     private final PaymentProviderService paymentProviderService;
 
+    public PaymentProviderController(PaymentProviderService paymentProviderService) {
+        this.paymentProviderService = paymentProviderService;
+    }
+
+    @PostMapping("/storefront/payment-providers")
+    public ResponseEntity<PaymentProviderVm> create(@Valid @RequestBody CreatePaymentVm createPaymentVm) {
+        return ResponseEntity
+            .status(HttpStatus.CREATED)
+            .body(paymentProviderService.create(createPaymentVm));
+    }
+
+   @PutMapping("/storefront/payment-providers")
+    public ResponseEntity<PaymentProviderVm> update(@Valid @RequestBody UpdatePaymentVm updatePaymentVm) {
+        return ResponseEntity.ok(paymentProviderService.update(updatePaymentVm));
+    }
+
     @GetMapping("/storefront/payment-providers")
-    public ResponseEntity<List<PaymentProviderVm>> getPaymentProviders() {
-        return ResponseEntity.ok(paymentProviderService.getEnabledPaymentProviders());
+    public ResponseEntity<List<PaymentProviderVm>> getAll(Pageable pageable) {
+        var paymentProviders = paymentProviderService.getEnabledPaymentProviders(pageable);
+        if (paymentProviders.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.NO_CONTENT).body(paymentProviders);
+        }
+        return ResponseEntity.ok(paymentProviders);
```

Comment:
Hi @duylv27, I noticed the new implementation returns 204 No Content for an empty list. My concern is that this might lead to an undefined body when the client is expecting JSON. Would it be better to return a 200 OK status with an empty array ([]) in the body instead?

![image](https://github.com/user-attachments/assets/9081e553-1b3f-4049-8fbb-edc989abedde)